exporters/stats/stackdriver/src/main/java/io/opencensus/exporter/stats/stackdriver/StackdriverStatsMonitoredResource.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright 2017, OpenCensus Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opencensus.exporter.stats.stackdriver;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import com.google.api.MonitoredResource;
+import javax.annotation.concurrent.GuardedBy;
+import javax.annotation.concurrent.ThreadSafe;
+
+/**
+ * {@code ThreadSafe} class that holds Stackdriver {@link MonitoredResource} for this {@link
+ * StackdriverStatsExporter}.
+ */
+@ThreadSafe
+final class StackdriverStatsMonitoredResource {
+
+  private static final Object monitor = new Object();
+
+  @GuardedBy("monitor")
+  private static MonitoredResource monitoredResource =
+      MonitoredResource.newBuilder().setType("global").build();
+
+  static void setMonitoredResource(MonitoredResource monitoredResource) {
+    checkNotNull(monitoredResource);
+    synchronized (monitor) {
+      StackdriverStatsMonitoredResource.monitoredResource = monitoredResource;
```

Comment:
MonitoredResource is not immutable so you need to make a copy.

Reply:
Done.

Reply:
I just realized that MonitoredResource is a protocol buffer, so it should be immutable.  See the "Builders" section in https://developers.google.com/protocol-buffers/docs/reference/java-generated.

Reply:
Oh you're right. Once it's built, its fields cannot be changed so should be immutable. I've removed the deep copy code.